pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoWriterTools.java

```diff
@@ -1556,8 +1556,10 @@ public static <T> int sizeOfMessageList(FieldDefinition field, List<T> list, ToI
      */
     public static int sizeOfBytesList(FieldDefinition field, List<? extends RandomAccessData> list) {
         int size = 0;
-        for (final RandomAccessData value : list) {
-            size += Math.toIntExact(sizeOfTag(field, WIRE_TYPE_DELIMITED) + sizeOfVarInt32(Math.toIntExact(value.length())) + value.length());
+        final int listSize = list.size();
+        for(int i = 0; i < listSize; i++) {
+            final long valueLength = list.get(i).length();
+            size += Math.toIntExact(sizeOfTag(field, WIRE_TYPE_DELIMITED) + sizeOfVarInt32(Math.toIntExact(valueLength)) + valueLength);
```

Comment:
It looks very similar (identical?) to `sizeOfDelimited()`, why not use this method here?

Reply:
Fixed